nautobot/extras/tasks.py

```diff
@@ -87,3 +92,78 @@ def provision_field(field_id, content_type_pk_set):
         for obj in model.objects.all():
             obj._custom_field_data[field.name] = field.default
             obj.save()
+
+
+@app.task()
+def process_webhook(webhook_pk, data, model_name, event, timestamp, username, request_id):
+    """
+    Make a POST request to the defined Webhook
+    """
+    from nautobot.extras.models import Webhook  # avoiding circular import
+
+    webhook = Webhook.objects.get(pk=webhook_pk)
+
+    context = {
+        "event": dict(ObjectChangeActionChoices)[event].lower(),
+        "timestamp": timestamp,
+        "model": model_name,
+        "username": username,
+        "request_id": request_id,
+        "data": data,
+    }
+
+    # Build the headers for the HTTP request
+    headers = {
+        "Content-Type": webhook.http_content_type,
+    }
+    try:
+        headers.update(webhook.render_headers(context))
+    except (TemplateError, ValueError) as e:
+        logger.error("Error parsing HTTP headers for webhook {}: {}".format(webhook, e))
+        raise e
+
+    # Render the request body
+    try:
+        body = webhook.render_body(context)
+    except TemplateError as e:
+        logger.error("Error rendering request body for webhook {}: {}".format(webhook, e))
+        raise e
+
+    # Prepare the HTTP request
+    params = {
+        "method": webhook.http_method,
+        "url": webhook.payload_url,
+        "headers": headers,
+        "data": body.encode("utf8"),
+    }
+    logger.info(
+        "Sending {} request to {} ({} {})".format(params["method"], params["url"], context["model"], context["event"])
+    )
+    logger.debug(params)
```

Comment:
Generally we shouldn't invoke logger with a non-string object as the first parameter.

```suggestion
    logger.debug("%s", params)
```